### PR TITLE
Upgrades

### PR DESCRIPTION
**Pull request by jankeromnes**
There are new versions of Ruby and Node.js.

A word of caution: I didn't test this, so it might break things down the line.

### CODE REVIEW

**Comment by svenefftinge** [comment]
Thanks for the contribution. Bumping node from 8 to 10 is not really a minor change.

**Comment by jankeromnes** [comment]
Ok you're right. It used to be a breaking change for Janitor, but Node 10 backwards compatibility seems to have improved lately because we were recently able to upgrade all our images without further changes or breakage.

But it should definitely be tested first. How can I do that?

**Comment by svenefftinge** [comment]
@akosyakov @AlexTugarev opinions?

File: full/Dockerfile [hunk 1]
```diff
@@ -70,7 +70,7 @@ RUN $HOME/.cargo/bin/rustup update \
     && $HOME/.cargo/bin/rustup component add rls-preview rust-analysis rust-src
 
 # Install nvm with node and npm
-ENV NODE_VERSION 8
+ENV NODE_VERSION 10
```
Comment (AlexTugarev) [inline full/Dockerfile]: Working on Theia requires node version 8. We'd need to create a custom image for Theia then, right?

Comment (jankeromnes) [reply full/Dockerfile]: It's true that upgrading to node 10 used to break Theia's install, but this has now been fixed. Are there any other Theia / node 10 blockers that I'm not aware of? We should probably test this, or maybe take it upstream.

Comment (AlexTugarev) [reply full/Dockerfile]: That's a good question. I've seen a lot of effort towards node 10 support, but I'm not sure how far this is.

I just did a quick test, and there is still [something to to on the Theia side](https://github.com/theia-ide/theia/issues/2009#issuecomment-446205981).

**Comment by orucky** [comment]
Isn't it that we have `nvm` there to let user use any version of Node.js he wants? In Theia (and other descending images) we can switch to some fixed version (currently 8.14.0, see #34), I'd also keep it there as a default in case someone wants to try Gitpod on Theia, but we can preinstall Node 10 there too, so the user can switch fast. Am I missing something? (That's well possible)

**Comment by svenefftinge** [comment]
The question is what should be set by default. It makes sense to use the latest stable which is Node 10.

File: full/Dockerfile (hunk 1 above)
Comment (AlexTugarev) [reply full/Dockerfile]: +1 for bumping. We can handle it in Theia's `.gitpod.yml`

**Comment by orucky** [comment]
One note - I bumped Ruby independently in #34. I know there's also a bit newer version of Python (3.6.7), Node.js is current (in 8.* line), there's Clang 7 instead of 6.0 (I'm creating an issue for this one as it's a major version). The rest I haven't checked

**Comment by jankeromnes** [comment]
Thanks all! This pull request is now superseded by #69 and #70.

**Pull request closed without merging**